src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/person/domain/OppfolgingstilfelleDag.kt

```diff
@@ -76,24 +81,34 @@ private fun List<OppfolgingstilfelleDag>.findLastPriorityOppfolgingstilfelleBit(
         it.priorityOppfolgingstilfelleBit != null
     }.priorityOppfolgingstilfelleBit
 
-fun List<OppfolgingstilfelleDag>.gradertAtTilfelleEnd() =
-    this.findLastPriorityOppfolgingstilfelleBit()?.isGradert() ?: false
+fun List<OppfolgingstilfelleDag>.allGradert() =
+    this.all { dag -> dag.priorityOppfolgingstilfelleBit?.isGradert() ?: false }
+
+fun List<OppfolgingstilfelleDag>.findDag(date: LocalDate) =
+    this.find { oppfolgingstilfelleDag -> oppfolgingstilfelleDag.dag == date }
 
-fun List<OppfolgingstilfelleDag>.toOppfolgingstilfelle(): Oppfolgingstilfelle {
+fun List<OppfolgingstilfelleDag>.toOppfolgingstilfelle(
+    oppfolgingstilfelleDagerPerVirksomhet: Map<String?, List<OppfolgingstilfelleDag>>,
+): Oppfolgingstilfelle {
     if (this.onlySykmeldingNyOrInntektsmelding() && this.durationDays() > 118) {
         val sampleUUID = this.firstNotNullOfOrNull { dag -> dag.priorityOppfolgingstilfelleBit }?.uuid
         log.info("Created oppfolgingstilfelle with duration>118 days based on only sykmelding-ny, bit sample uuid: $sampleUUID")
         SYKMELDING_NY_COUNTER.increment()
     }
     val arbeidstakerAtTilfelleEnd = this.isArbeidstakerAtTilfelleEnd()
+    val tilfelleEnd = this.last().dag
+    val allGradertAtTilfelleEnd = oppfolgingstilfelleDagerPerVirksomhet
+        .mapNotNull { it.value.findDag(tilfelleEnd) }
```

Comment:
Burde man heller bruke den siste dagen som finnes for hver enkelt virksomhet?

Reply:
Tenkte på det. Men er det relevant å sjekke om man er gradert potensielt tidlig i tilfellet? I mitt hode er det ved 8 uker eller på slutten av tilfellet som betyr noe, men jeg er neimen ikke sikker.